src/convenience.jl

```diff
@@ -0,0 +1,56 @@
+export read_csv_folder, show_tables, get_table
+
+"""
+    read_csv_folder(connection, folder)
+
+Read all CSV files in the `folder` and create a table for each in the `connection`.
+
+## Keywords arguments
+
+- `table_name_prefix = ""`
+- `table_name_suffix = ""`
+"""
+function read_csv_folder(connection, folder; table_name_prefix = "", table_name_suffix = "")
+    for filename in readdir(folder)
+        if !endswith(".csv")(filename)
+            continue
+        end
+        table_name, _ = splitext(filename)
+        table_name = replace(table_name, "-" => "_")
+        table_name = table_name_prefix * table_name * table_name_suffix
+        create_tbl(connection, joinpath(folder, filename); name = table_name)
+    end
+
+    return connection
+end
+
+"""
+    df = show_tables(connection)
+    query = show_tables(Val(:raw), connection)
+
+Run the `SHOW TABLES` sql command.
+
+The `Val(:raw)` variant returns the raw output from DuckDB, otherwise we construct a DataFrame.
+"""
+function show_tables(::Val{:raw}, connection)
+    DBInterface.execute(connection, "SHOW TABLES")
+end
+
+"""
+    df = get_table(connection, table_name)
+    query = get_table(Val(:raw), connection, table_name)
+
+Run the `SELECT * FROM table_name` sql command.
+
+The `Val(:raw)` variant returns the raw output from DuckDB, otherwise we construct a DataFrame.
+"""
+function get_table(::Val{:raw}, connection, table_name)
+    DBInterface.execute(connection, "SELECT * FROM $table_name")
+end
+
+for foo in (:show_tables, :get_table)
```

Comment:
Is this a general structure example to use later?

I mention it, because I don't see a test for this function `foo`

Reply:
This is metaprogramming. It's unnecessary for only these two functions, but it might be useful as it grows.
I only define the `Val{:foo}` functions above, then I use meta programming to define the other function, since the definition of both of them is the same: "Call the Val(:foo) variant and wrap in a DataFrame".

More details:
- `foo` loops over the other function names.
- `@eval` transforms `$foo` into the actual function name.
- The definition of `$foo` for an argument `(con, <whatever>)` is to call `DataFrame(<Val(:raw) version of $foo>)`.

Reply:
Awesome! I learnt something new. Thanks!